src/index.rs

```diff
@@ -1408,17 +1408,22 @@ impl Index {
     }
 
     /// returns the intrinsic (built-in) type represented by the given type-information
-    /// this will return the built-in type behind alias and range-types
+    /// this will return the built-in type behind alias / range-types / pointer-types
     pub fn find_intrinsic_type<'idx>(
         &'idx self,
         initial_type: &'idx DataTypeInformation,
     ) -> &'idx DataTypeInformation {
         match initial_type {
-            DataTypeInformation::SubRange { .. } | DataTypeInformation::Alias { .. } => {
+            DataTypeInformation::SubRange { .. }
+            | DataTypeInformation::Alias { .. }
+            | DataTypeInformation::Pointer { .. } => {
                 let inner_type_name = match initial_type {
                     DataTypeInformation::SubRange {
                         referenced_type, ..
                     } => referenced_type,
+                    DataTypeInformation::Pointer {
```

Comment:
I think you can combine this with the above line: 
```rust
DataTypeInformation::Subrange { referenced_type, ...} | DatatypeInformation::Pointer{inner_type_name : referenced_type,..} => referend_type,
```

Not sure of the order , but something similar should work